src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -467,10 +492,39 @@ private interface ILoader<TLeft, TRight>
             public static nuint Count128 => (uint)Vector128<T>.Count;
             public static nuint Count256 => (uint)Vector256<T>.Count;
             public static nuint Count512 => (uint)Vector512<T>.Count;
+            public static bool Use64Bit => IntPtr.Size == 8;
             public static Vector128<T> Load128(ref T ptr) => Vector128.LoadUnsafe(ref ptr);
             public static Vector256<T> Load256(ref T ptr) => Vector256.LoadUnsafe(ref ptr);
             public static Vector512<T> Load512(ref T ptr) => Vector512.LoadUnsafe(ref ptr);
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static bool EqualAndAscii64(ref T left, ref T right)
+            {
+                ulong leftValues = Unsafe.ReadUnaligned<ulong>(ref Unsafe.As<T, byte>(ref left));
+                ulong rightValues = Unsafe.ReadUnaligned<ulong>(ref Unsafe.As<T, byte>(ref right));
+
+                if (leftValues != rightValues || !AllCharsInUInt64AreAscii<T>(leftValues))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static bool EqualAndAscii128(ref T left, ref T right)
+            {
+                Vector128<T> leftValues = Vector128.LoadUnsafe(ref left);
+                Vector128<T> rightValues = Vector128.LoadUnsafe(ref right);
+
+                if (leftValues != rightValues || !AllCharsInVectorAreAscii(leftValues))
+                {
+                    return false;
+                }
+
+                return true;
```

Comment:
```suggestion
                return leftValues == rightValues && AllCharsInVectorAreAscii(leftValues);
```
I'd clean-up the existing method exits in this type here too. Codegen should be the same as before (and I can't remember why it was written that way).

Reply:
I thought so too, but this was actually intentional from the original author. A conditional return here is adding extra instructions.

The same code before and after:
```asm
kortestw  k1,k1
jne       near ptr M01_L11
vptest    ymm0,ymmword ptr [7FFC7E491580]
jne       near ptr M01_L11
```

```asm
kortestw  k1,k1
jae       near ptr M01_L00
vptest    ymm0,ymmword ptr [7FFC841B1600]
sete      r9b
movzx     r9d,r9b
test      r9d,r9d
je        near ptr M01_L00
```

I could add a comment here about it though. Any idea what issue is tracking this?

Reply:
> Any idea what issue is tracking this?

@EgorBo do you have any idea here? Can this be fixed in the JIT (as the natural way of writing the code produces more machine code).

Reply:
I think it's fine to prefer better code readability over such quirks as we constantly improve JIT over those